File: Ghpr.Core/Interfaces/ITestRun.cs
﻿using System;
using System.Collections.Generic;
using Ghpr.Core.Common;
using Ghpr.Core.Enums;

namespace Ghpr.Core.Interfaces
{
    public interface ITestRun
    {
        string Name { get; set; }
        string FullName { get; set; }
        double TestDuration { get; set; }
        string TestStackTrace { get; set; }
        string TestMessage { get; set; }
        string Result { get; set; }
        string TestType { get; set; }
        string Output { get; set; }
        string Priority { get; set; }
        string[] Categories { get; set; }
        ItemInfo TestInfo { get; set; }
        Guid RunGuid { get; set; }
        List<ITestScreenshot> Screenshots { get; set; }
        List<ITestEvent> Events { get; set; }
        List<ITestData> TestData { get; set; }

        TestResult TestResult { get; }
        bool FailedOrBroken { get; }
    }
}

File: Ghpr.Controller.ts/controller/interfaces/ITestRun.ts
///<reference path="./ITestScreenshot.ts"/>
///<reference path="./ITestData.ts"/>
///<reference path="./ITestEvent.ts"/>
///<reference path="./IItemInfo.ts"/>
///<reference path="./../enums/TestResult.ts"/>

interface ITestRun {
    name: string;
    fullName: string;
    description: string;
    duration: number;
    testInfo: IItemInfo;
    testStackTrace: string;
    testMessage: string;
    result: string;
    output: string;
    runGuid: string;
    testType: string;
    priority: string;
    categories: Array<string>;
    screenshots: Array<ITestScreenshot>;
    testData: Array<ITestData>;
    events: Array<ITestEvent>;

    testRunColor: string;
    testResult: TestResult;
}